src/mon/MDSMonitor.cc

```diff
@@ -1439,6 +1439,27 @@ bool MDSMonitor::prepare_command(MonOpRequestRef op)
   }
 }
 
+bool MDSMonitor::_has_health_warnings(vector<int> warnings)
+{
+  for (map<uint64_t, MDSHealth>::iterator i = pending_daemon_health.begin();
+       i != pending_daemon_health.end(); ++i) {
```

Comment:
```suggestion
  for (auto& [gid, health] : pending_daemon_health) {
```

Reply:
Please also update `pending_daemon_health` type to `std::map<mds_gid_t, MDSHealth>`. same key change for `pending_daemon_health_rm` too please.

Reply:
Thanks! Done.